src/core/maprenderer/qgsmaprendererjob.cpp

```diff
@@ -407,6 +424,18 @@ QPainter *QgsMapRendererJob::allocateImageAndPainter( QString layerId, QImage *&
   return painter;
 }
 
+QPainter *QgsMapRendererJob::allocatePictureAndPainter( QPicture *&picture )
```

Comment:
would be cleaner to render a std::tuple/pair of unique_ptrs here

Reply:
Done, but with a pair of pointer, because painter is only a pointer in QgsRenderContext